projects/lang_feedback/base_gen.py

```diff
@@ -0,0 +1,119 @@
+# Unalignment Generation
+from __future__  import annotations
+import json
+import os
+import argparse
+
+import sys
+from tqdm.auto import tqdm
+from vllm import SamplingParams,LLM
+PROMPT_USER: str = 'USER: {prompt}'
+PROMPT_ASSISTANT: str = '\nASSISTANT:'  # should not have a space at the end
+PROMPT_INPUT: str = PROMPT_USER + PROMPT_ASSISTANT
+from datasets import load_dataset, concatenate_datasets
+from PIL import Image
+import requests
+
+import os
+
+# 设置CUDA可见设备
+os.environ["CUDA_VISIBLE_DEVICES"] = "0, 1, 2, 3, 4, 5, 6, 7" 
+
+def parse_arguments() -> argparse.Namespace:
+    """Parse the command-line arguments."""
+    parser = argparse.ArgumentParser(
+        description='Evaluate models with gpt4',
+    )
+    # Model
+    parser.add_argument(
+       '--model_name_or_path',
+        type=str,
+        help='the name or path of the first model (champion) in the arena to load from',
+    )
+    parser.add_argument(
+        '--input_path',
+        type=str,
+        help='the path of the input json file',
+    )
+    parser.add_argument(
+        '--output_name',
+        type=str,
+        help='the name of the output json file',
+        default=None,
+    )
+    parser.add_argument(
+        '--output_dir',
+        type=str,
+        help='Where to store the eval output.',
+    )
+    return parser.parse_args()
+
+
+def generate_answer_by_vllim(problems: list[str], model_name_or_path:str) ->list[str]:
+    samplingparams = SamplingParams(
+        temperature = 1.0,
+        repetition_penalty = 1.1,
+        max_tokens = 2048,
+        n=1,
+    )
+    
+    llm = LLM(model=model_name_or_path, 
+              gpu_memory_utilization=0.95, 
+              swap_space=32, 
+              trust_remote_code=True, 
+              tensor_parallel_size=8)  # 使用8个GPU并行
```

Comment:
also

Reply:
Fixed both just now in https://github.com/PKU-Alignment/align-anything/pull/57/commits/ab64bb65c2364255eb46f81aa3feb19f96f53c7b